### PR TITLE
[DB] Fix reusing db sessions

### PR DESCRIPTION
**Pull request by liranbg**
https://jira.iguazeng.com/browse/ML-5424

ensured db session is created for when listing function hubs to avoid race condition accessing to db session

### CODE REVIEW

File: mlrun/run.py [hunk 1]
```diff
@@ -367,7 +367,7 @@ def import_function(url="", secrets=None, db="", project=None, new_name=None):
         if not runtime:
             raise KeyError(f"function {name}:{tag} not found in the DB")
     else:
-        url, is_hub_uri = extend_hub_uri_if_needed(url)
+        url, is_hub_uri = extend_hub_uri_if_needed(url, db)
```
Comment (alonmr) [inline mlrun/run.py:370]: seems like db is expected to be a string
```
:param db: optional, mlrun api/db path
```

Comment (liranbg) [reply mlrun/run.py:370]: few lines up it changes it to rundb db is a string

Comment (alonmr) [reply mlrun/run.py:370]: you are in the else clause

Comment (liranbg) [reply mlrun/run.py:370]: oh, right right

**Pull request merged**